packages/demo-saas/src/App.tsx

```diff
@@ -1,15 +1,84 @@
+import Loader from '@mui/material/CircularProgress';
+import Box from '@mui/material/Box';
+import { FronteggProvider } from '@frontegg/react';
+import { FronteggAppOptions } from '@frontegg/types';
+import { authOptions } from './customizationOptions/authOptions';
+import HomePage from './HomePage';
+import ModalsStepUpPage from './stepUp/ModalsStepUpPage';
+import HOCStepUpPage from './stepUp/HOCStepUpPage';
+import SimpleStepUpButtonPage from './stepUp/SimpleStepUpButtonPage';
+import SmallMaxAgeStepUpPage from './stepUp/SmallMaxAgeStepUpPage';
+import TransferStepUpPage from './stepUp/TransferStepUpPage';
+import NoMaxAgeStepUpPage from './stepUp/NoMaxAgeStepUpPage';
+import EntitlementsPage from './entitlements/EntitlementsPage';
+import Fallback from './Fallback';
+import NotAFronteggPage from './NotAFronteggPage';
+import { DEFAULT_BASE_URL } from './consts';
+import { ROUTE_PATHS } from './BaseHomePage/components/Links';
+
+const IS_HOSTED_LOGIN = true;
+
+const fronteggOptions: FronteggAppOptions =
+  // @ts-ignore
+  window.CYPRESS_CONFIG ||
+  ({
+    contextOptions: {
+      // baseUrl: process.env.PUBLIC_URL || process.env.REACT_APP_BASE_URL || DEFAULT_BASE_URL,
+      // clientId: process.env.REACT_APP_CLIENT_ID,
+    },
+
+    // guidesCdnUrl: 'https://assets.frontegg.com/admin-box/embedded-guides/1.0.0',
+    ...authOptions,
+    enableOpenAppRoute: true,
+  } as FronteggAppOptions);
+
+export const App: FC = () => {
+  const [loading, setLoading] = useState(true);
 
-export const App = () => {
-  console.log('App');
   return (
-    <div className='app-container'>
-      <AppHeader />
-      <AppSidebar />
-      <AppContent />
-    </div>
+    <BrowserRouter>
+      <FronteggProvider
+        {...fronteggOptions}
+        customLoader={setLoading}
+        entitlementsOptions={{ enabled: true }}
+        hostedLoginBox={IS_HOSTED_LOGIN} // don't remove it! change the flag above
+      >
+        <Switch>
+          <Route path={ROUTE_PATHS.HOME_PAGE} exact render={HomePage} />
+          <Route path={ROUTE_PATHS.ENTITLEMENTS} exact render={EntitlementsPage} />
+          <Route path={ROUTE_PATHS.STEP_UP_HIGH_MAX_AGE} exact render={SimpleStepUpButtonPage} />
+          <Route path={ROUTE_PATHS.STEP_UP_SMALL_MAX_AGE} exact render={SmallMaxAgeStepUpPage} />
+          <Route path={ROUTE_PATHS.STEP_UP_NO_MAX_AGE} exact render={NoMaxAgeStepUpPage} />
+          <Route path={ROUTE_PATHS.STEP_UP_MODALS} exact render={ModalsStepUpPage} />
+          <Route path={ROUTE_PATHS.STEP_UP_HOC} exact render={HOCStepUpPage} />
+          <Route path={ROUTE_PATHS.STEP_UP_TRANSFER} exact render={TransferStepUpPage} />
+          {/* For tests that use someurl as the authenticated-url */}
+          <Route path={'/someurl'} exact render={NotAFronteggPage} />
+          <Route
+            path={'/test'}
+            exact
+            component={() => {
+              return <div>Test</div>;
+            }}
+          />
```

Comment:
why do we need this route?

Reply:
we want to test fallback route. We had a lot of issues regarding it.. and it reflects a real customer app